src/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -642,7 +646,6 @@ private bool AcquireServerCredentials(ref byte[] thumbPrint, byte[] clientHello)
                     throw new AuthenticationException(SR.net_ssl_io_no_server_cert);
                 }
             }
-            // This probably never gets called as this is a client options delegate
             else if (_sslAuthenticationOptions.CertSelectionDelegate != null)
             {
```

Comment:
What is .NET Framework behavior when CertSelectionDelegate returns null? Do we fallback to ServerCertificate?
Or is ServerCertificate basically useless whenever user sets CertSelectionDelegate?

Reply:
From a response under my comment it seems like we should have a test documenting that - will be reviewing new version soon

Reply:
I agree with @karelz we should define `null` behavior but if this is what we did on full framework I'm fine with that on this PR

Reply:
I don't know what the behavior is on framework, but the behavior I pin in the test is the same on Core 2.0

Reply:
Let's make sure we have a test case for it.

Reply:
The new test in SslStreamToStreamTest verifies this behavior.